### ISSUE TITLE
Kepler Exporter pods fails to load eBPF modules

### ISSUE THREAD

**Issue by vprashar2929** [sustainable-computing-io/kepler-operator#79]
When trying to deploy the Kepler operand using Kepler-Operator via the operator hub the Kepler-exporter daemonset's fails to load eBPF modules due to which there is no metric value populated by the exporter.
Logs for reference:
```
I0703 10:02:13.833452 1 gpu.go:46] Failed to init nvml, err: could not init nvml: error opening libnvidia-ml.so.1: libnvidia-ml.so.1: cannot open shared object file: No such file or directory
I0703 10:02:13.846924 1 acpi.go:67] Could not find any ACPI power meter path. Is it a VM?
I0703 10:02:13.852766 1 exporter.go:151] Kepler running on version: 364c44f
I0703 10:02:13.852785 1 config.go:212] using gCgroup ID in the BPF program: true
I0703 10:02:13.852806 1 config.go:214] kernel version: 5.14
I0703 10:02:13.852824 1 exporter.go:171] EnabledBPFBatchDelete: true
I0703 10:02:14.022039 1 exporter.go:184] Initializing the GPU collector
I0703 10:02:14.022303 1 watcher.go:67] Using in cluster k8s config
chdir(/lib/modules/5.14.0-284.18.1.el9_2.x86_64/build): No such file or directory
I0703 10:02:14.154285 1 bcc_attacher.go:74] failed to attach the bpf program: <nil>
I0703 10:02:14.154300 1 bcc_attacher.go:143] failed to attach perf module with options [-DMAP_SIZE=10240 -DNUM_CPUS=4]: failed to attach the bpf program: <nil>, from default kernel source.
I0703 10:02:14.154311 1 bcc_attacher.go:158] failed to attach perf module with options [-DMAP_SIZE=10240 -DNUM_CPUS=4]: failed to attach the bpf program: <nil>, not able to load eBPF modules
I0703 10:02:14.154323 1 exporter.go:201] failed to start : failed to attach bpf assets: failed to attach perf module with options [-DMAP_SIZE=10240 -DNUM_CPUS=4]: failed to attach the bpf program: <nil>, not able to load eBPF modules
I0703 10:02:14.154418 1 exporter.go:228] Started Kepler in 301.664983ms
```
As per my observations(_Might be wrong_) what happens is when Kepler operand is deployed it also does MC changes enabling cgroups and kernel-devel. Since it takes time for MC to be in-effect before that daemonset pods get created and they fail to load the required modules because of that. 
Post restart of daemonset pods everything works fine. The exporter is able to load the modules and some metric value is seen.

**Quick question:** Shouldn't enabling MCs be handled at the time of operator installation and not on Kepler operand deployment?

**Comment by husky-parul** [comment]
@vprashar2929 You are on the right path. Once the nodes have restarted you can re check the logs and find that ebpf modules were loaded after MC installation and node restart were complete. However we are moving towards building kpler container with kernel source. See relate PR https://github.com/sustainable-computing-io/kepler/pull/733. With this changes we should need MC at all.

Refrence code:
https://github.com/sustainable-computing-io/kepler-operator/blob/v1alpha1/controllers/kepler_controller.go#L385
We are testing with no MC in `kepler:0.5`

**Comment by husky-parul** [comment]
Fixed in 0.5.0

**Comment by vprashar2929** [comment]
@husky-parul Is it fixed in the 0.5.0 release of Operator? I can see the related [issue](https://github.com/sustainable-computing-io/kepler-operator/issues/78) is still open
